workos/resources/list.py

```diff
@@ -147,16 +156,22 @@ class BaseWorkOsListResource(
 ):
     object: Literal["list"]
     data: List[ListableResource]
-    list_metadata: ListMetadata
+    list_metadata: Union[ListAfterMetadata, ListMetadata]
 
     list_method: Callable = Field(exclude=True)
     list_args: ListAndFilterParams = Field(exclude=True)
 
     def _parse_params(self):
-        fixed_pagination_params = {
-            "order": self.list_args["order"],
-            "limit": self.list_args["limit"],
-        }
+        fixed_pagination_params = cast(
+            # Type hints consider this a mismatch because it assume the dictionary is dict[str, int]
+            Dict[str, Union[int, str, None]],
+            {
+                "limit": self.list_args["limit"],
+            },
+        )
```

Comment:
Type hints were a bit "too smart". Since `self.list_args["limit"] is an `int` it considers the entire dictionary to be of type `Dict[str, int]`.